hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieLogFormatReader.java

```diff
@@ -49,37 +54,85 @@
-      boolean readMetadata) throws IOException {
+                        boolean readBlockLazily) throws IOException {
     this.inputStream = fs.open(logFile.getPath(), bufferSize);
     this.logFile = logFile;
     this.readerSchema = readerSchema;
-    this.readMetadata = readMetadata;
+    this.readBlockLazily = readBlockLazily;
+    this.currentLogPosition = inputStream.getPos();
+    this.reverseLogFilePosition = this.lastReverseLogFilePosition = fs.getFileStatus(logFile.getPath()).getLen();
+    addShutDownHook();
   }
 
   HoodieLogFormatReader(FileSystem fs, HoodieLogFile logFile, Schema readerSchema,
-      boolean readMetadata) throws IOException {
-    this(fs, logFile, readerSchema, DEFAULT_BUFFER_SIZE, readMetadata);
+                        boolean readBlockLazily) throws IOException {
+    this(fs, logFile, readerSchema, DEFAULT_BUFFER_SIZE, readBlockLazily);
+  }
+
+  HoodieLogFormatReader(FileSystem fs, HoodieLogFile logFile, Schema readerSchema) throws IOException {
+    this(fs, logFile, readerSchema, DEFAULT_BUFFER_SIZE, false);
   }
 
   @Override
   public HoodieLogFile getLogFile() {
     return logFile;
   }
 
+  /**
+   * Close the inputstream when the JVM exits
+   */
+  private void addShutDownHook() {
+    Runtime.getRuntime().addShutdownHook(new Thread() {
+      public void run() {
+        try {
+          inputStream.close();
+        } catch (Exception e) {
+          // fail silently for any sort of exception
+        }
+      }
+    });
+  }
+
+  // TODO : convert content and block length to long by using ByteBuffer, raw byte [] allows for max of Integer size
   private HoodieLogBlock readBlock() throws IOException {
-    // 2. Read the block type
-    int ordinal = inputStream.readInt();
-    Preconditions.checkArgument(ordinal < HoodieLogBlockType.values().length,
-        "Invalid block byte ordinal found " + ordinal);
-    HoodieLogBlockType blockType = HoodieLogBlockType.values()[ordinal];
 
-    // 3. Read the size of the block
-    int blocksize = inputStream.readInt();
+    int blocksize = -1;
+    int ordinal = -1;
+    HoodieLogBlockType blockType = null;
+    Map<HeaderMetadataType, String> header = null;
+
+    try {
+
+      if (isOldMagic()) {
+        // 1 Read the block type for a log block
+        ordinal = inputStream.readInt();
+
+        Preconditions.checkArgument(ordinal < HoodieLogBlockType.values().length,
+            "Invalid block byte ordinal found " + ordinal);
+        blockType = HoodieLogBlockType.values()[ordinal];
+
+        // 2 Read the total size of the block
+        blocksize = inputStream.readInt();
+      } else {
+        // 1 Read the total size of the block
+        blocksize = (int) inputStream.readLong();
+      }
+
+    } catch (Exception e) {
+      // An exception reading any of the above indicates a corrupt block
+      // Create a corrupt block by finding the next MAGIC marker or EOF
+      return createCorruptBlock();
```

Comment:
This is already handled : https://github.com/uber/hudi/pull/331/files#diff-63c57738a6216c1e54d05c9dfc8af054L190

Reply:
can you expand on this more? which line handles which issue?

Reply:
This line : https://github.com/uber/hudi/pull/331/files#diff-63c57738a6216c1e54d05c9dfc8af054L190 returns a true for readMagic() if it either 1) find a magicHeader or 2) find EOF. So for the last good block, EOF handles the case and since readMagic() is called in isBlockCorrupt() and createCorruptBlock().
Not sure what's the other issue you're referring to ?